### PR TITLE
Upgrade lance to 0.5.5, and plumb thru new features from the upgrade

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
* upgrade
* fixes for the upgrade
* allow JS users to pass custom AWS credentials

TODO:
add JS code to actually allow passing custom creds, right now only the FFI allows custom creds

### CODE REVIEW

File: rust/ffi/node/src/lib.rs [hunk 1]
```diff
@@ -49,6 +53,33 @@ struct JsTable {
 
 impl Finalize for JsTable {}
 
+// TODO: object_store didn't export this type so I copied it.
+// Make a requiest to object_store to export this type
+#[derive(Debug)]
+pub struct StaticCredentialProvider<T> {
```
Comment (eddyxu) [inline rust/ffi/node/src/lib.rs:59]: Why this is in nodejs ffi?

File: rust/ffi/node/src/arrow.rs [hunk 2]
```diff
@@ -25,10 +24,13 @@ use lance::arrow::{FixedSizeListArrayExt, RecordBatchExt};
 pub(crate) fn convert_record_batch(record_batch: RecordBatch) -> RecordBatch {
     let column = record_batch
         .column_by_name("vector")
+        .cloned()
         .expect("vector column is missing");
-    let arr = as_list_array(column.deref());
+    // TODO: we should just consume the underlaying js buffer in the future instead of this arrow around a bunch of times
```
Comment (eddyxu) [inline rust/ffi/node/src/arrow.rs:29]: pls file an issue here

File: rust/ffi/node/src/lib.rs [hunk 3]
```diff
@@ -97,19 +128,74 @@ fn database_table_names(mut cx: FunctionContext) -> JsResult<JsPromise> {
     Ok(promise)
 }
 
+fn get_aws_creds<T>(
+    cx: &mut FunctionContext,
+    arg_starting_location: i32,
+) -> Result<Option<AwsCredentialProvider>, NeonResult<T>> {
+    let secret_key_id = cx
```
Comment (eddyxu) [inline rust/ffi/node/src/lib.rs:135]: This expects js side to pass these parameters. 
What would be js signature look like.

File: rust/ffi/node/src/lib.rs [hunk 4]
```diff
@@ -241,20 +327,29 @@ fn table_create(mut cx: FunctionContext) -> JsResult<JsPromise> {
...
-    let mut params = WriteParams::default();
-    params.mode = mode;
 
     let rt = runtime(&mut cx)?;
     let channel = cx.channel();
 
     let (deferred, promise) = cx.promise();
     let database = db.database.clone();
 
+    let aws_creds = match get_aws_creds(&mut cx, 3) {
```
Comment (eddyxu) [inline rust/ffi/node/src/lib.rs:337]: How to handle non-AWS credentials here? i.e., GCP and etc.

File: rust/ffi/node/src/lib.rs (hunk 1 above)
Comment (chebbyChefNEQ) [reply rust/ffi/node/src/lib.rs:59]: This shouldn't I'll make a PR to object store and remove this once object store exports this.

**Review by eddyxu — APPROVED** [review]

**Pull request merged**